lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -99,6 +101,31 @@ def batch_create_remaining_prospects
     [submissions, errors]
   end
 
+  # Deletes a list of prospects from Pardot. For Pardot API documentation, see
+  # http://developer.pardot.com/kb/api-version-4/prospects/#using-prospects.
+  # @param [Array[Integer]] prospect_ids of the prospects to be deleted.
+  # @return [Array[Integer]] the set of prospect_ids that failed to be deleted.
+  def self.delete_prospects(prospect_ids)
+    failed_prospect_ids = []
+
+    return prospect_ids unless CDO.rack_env? :production
+    return prospect_ids if prospect_ids.length > MAX_PROSPECT_DELETION_BATCH_SIZE
```

Comment:
will we get into issues by rejecting batches that are too big? Is there a reason we can't batch the deletes in some way?

Reply:
Other options to consider here, depends on how cautious we are.
1. Fire an Honeybadger error and return right here so someone can intervene and verify the deletion requests are correct.
2. Continue with the deletion but fire an Honeybadger error to notice the team after the fact.

Reply:
3. Continue with the deletion but only delete up to the daily limit. It may takes several days to completely delete all the requested prospects. Meanwhile, fire an Honeybadger error so we know that there is a higher-than-normal number of requests and intervene if necessary.

Reply:
Throttling will be addressed in another PR.